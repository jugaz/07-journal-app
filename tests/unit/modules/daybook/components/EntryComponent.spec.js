import { shallowMount } from "@vue/test-utils";
import Entry from '@/modules/daybook/components/EntryComponent'
import { journalState } from "../../../mock-data/test-journal-state";

describe('Pruebas en el Entry component',() => {

    const mockRouter = {
        push: jest.fn()
    }
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    });

    test('debe de hacer match con el snapshot',() => {

        expect( wrapper.html() ).toMatchSnapshot()
        
    })
    

    test('debe de redireccionar al hacer click en el entry-container',() => {

        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith({
            name: 'entry', 
            params: {
                id: journalState.entries[0].id
            }
        })
        

    })


    
    test('pruebas en las propiedads computadas',() => {
        
        expect( wrapper.vm.day ).toBe(23)
        expect( wrapper.vm.month ).toBe('Julio')
        expect( wrapper.vm.yearDay ).toBe('2021, Viernes')

    })

})