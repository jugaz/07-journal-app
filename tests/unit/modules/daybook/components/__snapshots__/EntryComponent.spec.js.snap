// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">16</span><span class="mx-1 fs-5">Mayo</span><span class="mx-2 fw-light">2022, Lunes</span></div>
  <div class="entry-description">Desde Firebase</div>
</div>
`;
